### PR TITLE
MueLu: Adding filtering support for BrickAggregation

### PR DESCRIPTION
**Pull request by csiefer2**
Issue: #10439
Tests:GCC
Stakeholder Feedback: As the stakeholder, I approve

### CODE REVIEW

File: packages/muelu/src/Graph/BrickAggregation/MueLu_BrickAggregationFactory_decl.hpp [hunk 1]
```diff
@@ -59,18 +59,20 @@
...
+#include "MueLu_LWGraph_fwd.hpp"
 #include "MueLu_Level_fwd.hpp"
 #include "MueLu_Aggregates_fwd.hpp"
 #include "MueLu_Exceptions.hpp"
 #include "MueLu_Utilities_fwd.hpp"
 
 /*!
   @class BrickAggregationFactory
-  @brief Aggregation method for generating "brick" aggregates.
+  @brief Aggregation method for generating "brick" aggregates.  It also does "hotdogs" and "pancakes."
```
Comment (jhux2) [inline packages/muelu/src/Graph/BrickAggregation/MueLu_BrickAggregationFactory_decl.hpp:72]: mmmm, pancakes.

File: packages/muelu/src/Graph/BrickAggregation/MueLu_BrickAggregationFactory_def.hpp [hunk 2]
```diff
@@ -480,6 +482,106 @@ namespace MueLu {
...
+      bool drop_x = (bx_ == 1);
+      bool drop_y = (nDim_> 1 && by_ == 1);
+      bool drop_z = (nDim_> 2 && bz_ == 1);
+      
+      ArrayRCP<LO> rows   (A->getLocalNumRows()+1);
+      ArrayRCP<LO> columns(A->getLocalNumEntries());
+      
+      size_t N = A->getRowMap()->getLocalNumElements();
+
+      // FIXME: Do this on the host because indexing functions are host functions
```
Comment (jhux2) [inline packages/muelu/src/Graph/BrickAggregation/MueLu_BrickAggregationFactory_def.hpp:526]: Does this imply data transfer to host is required?

**Review by jhux2 — COMMENTED** [review]
Is there anyway to unit test this?

File: packages/muelu/src/Graph/BrickAggregation/MueLu_BrickAggregationFactory_def.hpp (hunk 2 above)
Comment (csiefer2) [reply packages/muelu/src/Graph/BrickAggregation/MueLu_BrickAggregationFactory_def.hpp:526]: Well, BrickAggregation never did work on-device anyway, so my level of caring is low.

File: packages/muelu/src/Graph/BrickAggregation/MueLu_BrickAggregationFactory_decl.hpp (hunk 1 above)
Comment (csiefer2) [reply packages/muelu/src/Graph/BrickAggregation/MueLu_BrickAggregationFactory_decl.hpp:72]: German Apple Pancake.  Do it.  You will not regret.

**Review by jhux2 — APPROVED** [review]
lgtm

File: packages/muelu/utils/misc/autoretester.sh [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+        # Add AT: RETEST
+        CMD=$(echo curl -i -X POST -H \'Authorization: token $token\' -d \'{\"labels\": [\"AT: RETEST\"]}\' https://api.github.com/repos/$fork/$repo/issues/$pr/labels)
+        eval $CMD > $TMPFILE 2> /dev/null
+    fi
+
+    # Sleep for an hour
+    sleep 1800
+    
+    # Check to see if we're merged
+    if grep 'Status: 204' $TMPFILE > /dev/null; then
```
Comment (cgcgcg) [inline packages/muelu/utils/misc/autoretester.sh:90]: Shouldn't there be a request after the sleep but before the merge check?

Comment (csiefer2) [reply packages/muelu/utils/misc/autoretester.sh:90]: Yes.  Good catch

**Review by cgcgcg — APPROVED** [review]

**Pull request merged**